text/0049-network-ids.md

```diff
@@ -0,0 +1,183 @@
+  we are using server-side configuration to make all the token ids different on every network, arbitrarily, for "safety", but then we are hard-coding logic to basically
+  undo that per-network mangling into each client so that we can test the clients properly.
+* Ultimately, in a product-focused project, we want to be able to see in testing that "GOOG" displays correctly in the app when we test on testnet, and that all the correct
+  UI elements work, and token-specific config is loaded correctly. This will likely motivate engineers higher up in the stack to make "TESTGOOG" and "GOOG" display in identical ways
+  to the user of the app, so that they will know exactly how it will appear in real life in prod when they are testing in testnet.
+  This completely defeats the purpose "separate token ids per network" mechanism, since the user will no longer be able to distinguish being on prod vs. being on testnet.
+
+We propose to reduce complexity here and separate these concerns.
+
+* "Network id"-aware clients is a common pattern used by other blockchains to help clients (and users) be aware of what network they are talking to.
+* Network id can be easily displayed to the user if desired, which is something that none of our clients actually do today.
+   * For example, when using Metamask, a user can select whether to talk to mainnet, an L2 network, or a test network, via a dropdown menu, and see what network they are currently talking to here.
+     Even if we don't support dynamic switching of networks in a particular client, we can still display the network id somewhere, so that the users can discover TEST vs. PROD easily, without
+     changing any of the other codepaths that we want to test.
+   * This could be as simple as, if the network id is not prod, then the network id is displayed in a textbox in the upper right corner.
+   * So instead of "TESTGOOG", you still see the ticker "GOOG", but "TEST" appears in the user interface.
+* All of our dev, test, and prod environments currently have names that can naturally be adopted for this purpose.
+* Network id can be our primary method of "de-conflicting" and preventing mixing of networks.
+* We won't have to solve this again every time we add a new feature to consensus, and we can simplify future designs by reducing requirements. 
+* This has the benefit that the error messages when networks are mixed will be consistent and much clearer.
+
+Note that while we are partly motivated by reducing complexity and easing testing around MCIP #45, it is possible that we will do MCIP #45 as originally proposed
+and this proposal -- they are orthogonal, strictly speaking. We could decide to adopt this proposal as a secondary layer of defense against mixing problems, or in service
+of a new consensus feature in the future.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+A `network id` is a UTF-8 string. No particular limit on the length is imposed.
```

Comment:
We have some inconsistent naming around how we document the networks and how we label them in operations.

- prod vs MainNet
- test vs TestNet

We have a lot of hard to change (OK very inconvenient to change) pointers like dns, s3 buckets, grafana and sentry ids that use "prod", "test"... as identifiers. 

Should we make these `network_id` identifiers match them and change our external documentation to match. 

I think it might be confusing to users if they expect MainNet/TestNet and see IDs like "test" and "prod"

Reply:
One thing I didn't do in this doc yet is propose to standardize any particular network id strings for mainnet or testnet.

If you have specific proposals would love to take them.